src/mlpack/core/data/split_data.hpp

```diff
@@ -368,6 +374,228 @@ Split(const arma::Mat<T>& input,
                          std::move(testData));
 }
 
+/**
+ * Given an input dataset and labels, split into a training set and test set.
```

Comment:
Can you clarify the intention here? Are we aiming to split the `field` itself into a training and test set, or are we splitting each of the things held by the field into a training and test set?

Reply:
(I'm sure I'll have more comments but I need to understand this method a little better first. :))

Reply:
The intuition here is that, a user would be using `arma::field` only when the sizes of each data point is different. So, the assumption here is that the user would provide us an `arma::field (n_rows = 1, n_cols = n_cols)` where each of the element of the `row` inside the `field` would be a single data point and here we are trying to split that into two `arma::field`s each having a single `row` and `columns` according to the `testRatio`.
I hope that clarifies. Forgive me if I didn't make sense here.

Reply:
In the implementation, I have ignored the `n_slices` for the `arma::field`. But while writing the above comment, I got another idea that we could possibly assume the input data could be `arma::field (n_rows = 1, n_cols = 1, n_slices = # of data points)`. Thus we could assume that each `slice` of the `field` corresponds to a single data point and then we could split over those slices.
For me both are fine, and both would serve well in all purposes.
I think it is just a matter of what way people would like to store their data.
Please let me know your thoughts about it @rcurtin :)

Reply:
Hey there @RishabhGarg108, sorry for the slow response!  I think that the way you described is just fine---but can you update the documentation so that it's clear what the inputs and outputs are?  Thanks. :+1: